### ISSUE TITLE
Rule proposal: Enforce standard error constructors when subclassing (message, options)

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#1810]
### Description

`Error#cause` was recently added and the native way (Node 16.9+) to set this property is via the options parameter:

```js
new Error('Sup', {
	cause: new Error('Network failed')
})
```

When subclassing Error, this parameter can be lost or simply use a different format.

### Fail

```js
class OutOfBounds extends Error {
	constructor(message) {
		super('Oops: ', message)
	}
}
```

```js
class OutOfBounds extends Error {
	constructor(message, details) {
		super(message)
		this.details = details;
	}
}
```

### Pass

```js
class OutOfBounds extends Error {
	constructor(message, options) {
		super('Oops: ', message, options)
	}
}
```

```js
class OutOfBounds extends Error {
	constructor(message, options) {
		super(message, options)
		this.details = options?.details;
	}
}
```

**Comment by sindresorhus** [comment]
Maybe it should be part of [`custom-error-definition`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/custom-error-definition.md)?

**Comment by fregante** [comment]
Sounds like exactly it!

**Comment by sindresorhus** [comment]
Accepted
